OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -326,13 +325,15 @@ public MovePart(Move move, WPos from, WPos to, int fromFacing, int toFacing, int
 
 			public override bool Cancel(Actor self, bool keepQueue = false)
 			{
-				Move.Cancel(self, keepQueue);
-				return base.Cancel(self);
+				// MovePart has no child or queued activity so it is correct to cancel this
+				// then return the RootActivity's cancel result.
+				base.Cancel(self);
+				return RootActivity.Cancel(self, keepQueue);
```

Comment:
I don't think it is correct to cancel the whole queue. You don't know from which level of the tree the `Cancel` originates from. It could be from an activity that decides it wants to do something else instead of moving, but still continue running itself. This line would cancel that activity.